src/CoreBundle/Controller/CourseController.php

```diff
@@ -600,6 +604,132 @@ public function welcome(Course $course): Response
         ]);
     }
 
+    #[Route('/{id}/getToolIntro', name: 'chamilo_core_course_gettoolintro')]
+    public function getToolIntro(Request $request, Course $course, EntityManagerInterface $em): Response
+    {
+        $sessionId = (int) $request->get('sid');
+        error_log('Session id -> '.$sessionId);
+
+        //$session = $this->getSession();
+        $responseData = [];
+        $ctoolRepo = $em->getRepository(CTool::class);
+        $sessionRepo = $em->getRepository(Session::class);
+        $createInSession = false;
+
+        $session = null;
+        if (!empty($sessionId)) {
+            $session = $sessionRepo->find($sessionId);
+        }
+
+        $qb = $em->createQueryBuilder();
+        $query = $qb->select('ct')
+            ->from('Chamilo\CourseBundle\Entity\CTool', 'ct')
+            ->where('ct.course = :c_id')
+            ->andWhere('ct.name = :name')
+            ->andWhere(
+                $qb->expr()->orX(
+                    $qb->expr()->eq('ct.session', ':session_id'),
+                    $qb->expr()->isNull('ct.session')
+                )
+            )
+            ->setParameters([
+                'c_id' => $course->getId(),
+                'name' => 'course_homepage',
+                'session_id' => 0
+            ])
+            ->getQuery();
+
+        $ctool = $query->getOneOrNullResult();
+
+        if ($session) {
+            error_log('Session actual -> '.$session->getId());
+            error_log('Course actual -> '.$course->getId());
+            $ctoolSession = $ctoolRepo->findOneBy(['name' => 'course_homepage', 'course' => $course, 'session' => $session]);
+            if (!$ctoolSession) {
+                $createInSession = true;
+            } else {
+                $createInSession = false;
+                $ctool = $ctoolSession;
+            }
+        }
+
+
+        error_log('$ctool -> '.$ctool->getIid());
+        error_log('$createInSession -> '.$createInSession);
+
+        if ($ctool) {
```

Comment:
Si no se encuentra un $cTool puedes lanzar un NotFoundException

`return $this->createNotFoundException(`...